### PR TITLE
Add`.png` file previewer page

### PR DESCRIPTION
**Pull request by williamputraintan**
- Added new route `/files/{subjectId}` for png preview files from S3
- Available preview options (with RegExr): 
```
  { name: 'GPL - all', regexKey: 'gridss_purple_linx/(.*)(.png$)' },
  { name: 'GPL - purple', regexKey: 'gridss_purple_linx/purple/(.*)(.png$)' },
  { name: 'GPL - linx', regexKey: 'gridss_purple_linx/linx/(.*)(.png$)' },
  { name: 'UMCCRISED - all', regexKey: 'umccrised/(.*)(.png$)' },
```
- Added button to link to this routes via `lims-table` and `subject-page`

### CODE REVIEW

File: src/containers/FileViewers.tsx [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+};
+
+const GPL_PIPELINE_OPTIONS: pipelineOptionType[] = [
+  { name: 'GPL - all', regexKey: 'gridss_purple_linx/(.*)(.png$)' },
+  { name: 'GPL - purple', regexKey: 'gridss_purple_linx/purple/(.*)(.png$)' },
+  { name: 'GPL - linx', regexKey: 'gridss_purple_linx/linx/(.*)(.png$)' },
+];
+
+const UMCCRISED_PIPELINE_OPTIONS: pipelineOptionType[] = [
+  { name: 'UMCCRISED - all', regexKey: 'umccrised/(.*)(.png$)' },
```
Comment (victorskl) [inline src/containers/FileViewers.tsx:43]: Can you change display to just `umccrise` without `d`, here pls.

File: src/containers/FileViewers.tsx [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+        </Grid>
+      </Grid>
+    </Paper>
+  );
+}
+
+async function getPresignedUrlForEveryItem(apiResponse: any) {
+  for (const item of apiResponse.results) {
+    const id = item.id;
+    const presigned_url = await getPreSignedUrl(id);
```
Comment (victorskl) [inline src/containers/FileViewers.tsx:375]: Can we lazy fetch the PreSignedUrl request here, pls? i.e. Just grab one for page entry rendering purpose. Then, only when, user started clicking around next item then make a request and, push into local state for caching/reusing, for example.

File: src/containers/Home.js [hunk 3]
```diff
@@ -202,6 +204,15 @@ class Home extends Component {
                           </Link>
                         )}
                       </TableCell>
+                    ) : col.key === 'files_viewer' ? (
+                      <TableCell key={col.key}>
+                        <IconButton
+                          component={RouterLink}
+                          aria-label='files_viewer'
+                          to={`/files/${row.subject_id}`}>
```
Comment (victorskl) [inline src/containers/Home.js:212]: Can you hide or disabled this component / linking, if `subject_id` is null or undefined, pls. There are sample without Subject ID...

**Review by victorskl — COMMENTED** [review]
Looking good, Will. Fantastic! Couple of comments, pls see if you could tackle those. Happy discuss otherwise.

File: src/containers/FileViewers.tsx (hunk 2 above)
Comment (victorskl) [reply src/containers/FileViewers.tsx:375]: And, pls use `CircularProgress` or appropriate component for transition effect, if any.

File: src/containers/FileViewers.tsx (hunk 1 above)
Comment (williamputraintan) [reply src/containers/FileViewers.tsx:43]: Resolved in commit `bd8dc4a`

File: src/containers/Home.js (hunk 3 above)
Comment (williamputraintan) [reply src/containers/Home.js:212]: Resolved in commit `bd8dc4a`

File: src/containers/FileViewers.tsx (hunk 2 above)
Comment (victorskl) [reply src/containers/FileViewers.tsx:375]: (Hehe, don't hate me! LoL) If it gets tedious to do, let me know. We can try another option/idea. @williamputraintan

Comment (williamputraintan) [reply src/containers/FileViewers.tsx]: > And, pls use CircularProgress or appropriate component for transition effect, if any.

When using this function, it is sandwiched with [isLoading state ](https://github.com/umccr/data-portal-client/blob/fa7089df0910417980e3d5795a89919b6a3c2817/src/containers/FileViewers.tsx#L92)that have a CircularProgress component.

Comment (williamputraintan) [reply src/containers/FileViewers.tsx]: >  don't hate me

Nope, not hating you! 😅 _(It should be this way in the beginning)_

> We can try another option/idea.

Did another push to resolve this 😉

Comment (victorskl) [reply src/containers/FileViewers.tsx]: Great! I'll check out in tick.

**Review by victorskl — APPROVED** [review]
Awesomeness, Will! Let's push it to DEV and UAT, next.

**Pull request merged**